pkg/types/datum_pool.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import "github.com/pingcap/tidb/pkg/util/zeropool"
+
+var (
+	defaultDatumSliceSize = 64
+	datumPool             = zeropool.New[[]Datum](func() []Datum {
+		return make([]Datum, 0, defaultDatumSliceSize)
+	})
+)
+
+func allocDatum(n int) []Datum {
+	if n > defaultDatumSliceSize {
```

Comment:
if `n > defaultDatumSliceSize` it will return a datum array with length `n`, but if `n <= defaultDatumSliceSize`, it will return a datum array with length `0`, is it the expected behavior?